File: Backend/innkt.StringLibrary/Models/Language.cs
using System.ComponentModel.DataAnnotations;

namespace innkt.StringLibrary.Models;

/// <summary>
/// Represents a supported language in the system
/// </summary>
public class Language
{
    [Key]
    public int Id { get; set; }
    
    /// <summary>
    /// Language code (e.g., "en", "es", "fr", "de", "ro")
    /// </summary>
    [Required]
    [MaxLength(10)]
    public string Code { get; set; } = string.Empty;
    
    /// <summary>
    /// Language name in English (e.g., "English", "Spanish", "French", "German", "Romanian")
    /// </summary>
    [Required]
    [MaxLength(50)]
    public string Name { get; set; } = string.Empty;
    
    /// <summary>
    /// Native language name (e.g., "English", "Español", "Français", "Deutsch", "Română")
    /// </summary>
    [Required]
    [MaxLength(50)]
    public string NativeName { get; set; } = string.Empty;
    
    /// <summary>
    /// Whether this language is active/available
    /// </summary>
    public bool IsActive { get; set; } = true;
    
    /// <summary>
    /// Whether this is the default language
    /// </summary>
    public bool IsDefault { get; set; } = false;
    
    /// <summary>
    /// Language direction (LTR or RTL)
    /// </summary>
    [MaxLength(3)]
    public string Direction { get; set; } = "LTR";
    
    /// <summary>
    /// Creation timestamp
    /// </summary>
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    /// <summary>
    /// Last update timestamp
    /// </summary>
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: Mobile/innkt.Mobile/src/i18n/language.ts
export type Language = 
  | 'en'  // English
  | 'ro'  // Romanian
  | 'he'  // Hebrew
  | 'ar'  // Arabic
  | 'es'  // Spanish
  | 'fr'  // French
  | 'de'  // German
  | 'it'  // Italian
  | 'pt'  // Portuguese
  | 'ru'  // Russian
  | 'zh'  // Chinese
  | 'ja'  // Japanese
  | 'ko'  // Korean
  | 'hi'  // Hindi
  | 'tr'; // Turkish

export interface LanguageConfig {
  code: Language;
  name: string;
  nativeName: string;
  isRTL: boolean;
  dateFormat: string;
  numberFormat: string;
}

export const SUPPORTED_LANGUAGES: Record<Language, LanguageConfig> = {
  en: {
    code: 'en',
    name: 'English',
    nativeName: 'English',
    isRTL: false,
    dateFormat: 'MM/DD/YYYY',
    numberFormat: 'en-US',
  },
  ro: {
    code: 'ro',
    name: 'Romanian',
    nativeName: 'Română',
    isRTL: false,
    dateFormat: 'DD.MM.YYYY',
    numberFormat: 'ro-RO',
  },
  he: {
    code: 'he',
    name: 'Hebrew',
    nativeName: 'עברית',
    isRTL: true,
    dateFormat: 'DD/MM/YYYY',
    numberFormat: 'he-IL',
  },
  ar: {
    code: 'ar',
    name: 'Arabic',
    nativeName: 'العربية',
    isRTL: true,
    dateFormat: 'DD/MM/YYYY',
    numberFormat: 'ar-SA',
  },
  es: {
    code: 'es',
    name: 'Spanish',
    nativeName: 'Español',
    isRTL: false,
    dateFormat: 'DD/MM/YYYY',
    numberFormat: 'es-ES',
  },
  fr: {
    code: 'fr',
    name: 'French',
    nativeName: 'Français',
    isRTL: false,
    dateFormat: 'DD/MM/YYYY',
    numberFormat: 'fr-FR',
  },
  de: {
    code: 'de',
    name: 'German',
    nativeName: 'Deutsch',
    isRTL: false,
    dateFormat: 'DD.MM.YYYY',
    numberFormat: 'de-DE',
  },
  it: {
    code: 'it',
    name: 'Italian',
    nativeName: 'Italiano',
    isRTL: false,
    dateFormat: 'DD/MM/YYYY',
    numberFormat: 'it-IT',
  },
  pt: {
    code: 'pt',
    name: 'Portuguese',
    nativeName: 'Português',
    isRTL: false,
    dateFormat: 'DD/MM/YYYY',
    numberFormat: 'pt-PT',
  },
  ru: {
    code: 'ru',
    name: 'Russian',
    nativeName: 'Русский',
    isRTL: false,
    dateFormat: 'DD.MM.YYYY',
    numberFormat: 'ru-RU',
  },
  zh: {
    code: 'zh',
    name: 'Chinese',
    nativeName: '中文',
    isRTL: false,
    dateFormat: 'YYYY/MM/DD',
    numberFormat: 'zh-CN',
  },
  ja: {
    code: 'ja',
    name: 'Japanese',
    nativeName: '日本語',
    isRTL: false,
    dateFormat: 'YYYY/MM/DD',
    numberFormat: 'ja-JP',
  },
  ko: {
    code: 'ko',
    name: 'Korean',
    nativeName: '한국어',
    isRTL: false,
    dateFormat: 'YYYY. MM. DD.',
    numberFormat: 'ko-KR',
  },
  hi: {
    code: 'hi',
    name: 'Hindi',
    nativeName: 'हिन्दी',
    isRTL: false,
    dateFormat: 'DD/MM/YYYY',
    numberFormat: 'hi-IN',
  },
  tr: {
    code: 'tr',
    name: 'Turkish',
    nativeName: 'Türkçe',
    isRTL: false,
    dateFormat: 'DD.MM.YYYY',
    numberFormat: 'tr-TR',
  },
};

export const getLanguage = (language: Language): LanguageConfig => {
  return SUPPORTED_LANGUAGES[language] || SUPPORTED_LANGUAGES.en;
};

export const isRTL = (language: Language): boolean => {
  return SUPPORTED_LANGUAGES[language]?.isRTL || false;
};






